### PR TITLE
Lodash: Refactor away from `_.words()`

### PR DESCRIPTION
**Pull request by tyxla**
## What?
This PR removes the `_.words()` usage completely and deprecates the function. It also removes one of the last remaining `_.deburr()` usages in the same file in favor of `remove-accents` that we've been using in other packages already.

## Why?

Lodash is known to unnecessarily inflate the bundle size of packages, and in most cases, it can be replaced with native language functionality. See these for more information and rationale:

* https://github.com/WordPress/gutenberg/issues/16938#issuecomment-602837246
* https://github.com/WordPress/gutenberg/issues/17025
* https://github.com/WordPress/gutenberg/issues/39495 

## How?

Similar to #42427, we're suggesting to use the `change-case` library, which offers modular functions for all the casing functions we use from Lodash (like `snakeCase`, `capitalize`, `startCase`, `camelCase`, `kebabCase` ), at a small price (the methods are small themselves), and has TS support. The benefit of using that external library for all those case conversion functions is that we won't have to maintain our own versions of them, and those are already broadly used and tested.

In particular, we replace the single instance of `words()` with a custom implementation that uses `noCase()` to generate a normalized string of words, then splits it into an array and filters out the empty values.

We also use the opportunity to remove one of the last remaining `_.deburr()` usages in that same file, since it touches the same functionality we're testing here. 

## Testing Instructions
* Verify all tests still pass, particularly:
  * `npm run test-unit packages/block-editor/src/components/inserter/test/search-items.js`
* Verify searching in all inserters (inline slash inserter, header toolbar, inline block editor inserter) still works well:
  * Searching for blocks using their name, description or keywords still works the same way as before.
  * Searching for a string with diacritics still works the same way: `média`

### CODE REVIEW

File: packages/block-editor/src/components/inserter/search-items.js
Comment (talldan) [inline packages/block-editor/src/components/inserter/search-items.js:4]: `noCase` could be overkill given the string being operated on is human search text. Seems like the main thing it does is insert spaces into camel cased text, but I don't expect humans will write that way (even those of us that are coders 😄).

edit - ah, I missed that this is also used on the block name, github had collapsed that part of the code, you can ignore this comment!

**Comment by gziolo** [comment]
**Not a blocker for this PR**, but a general note to take into account so we can address it in a systematic way.

> It also removes one of the last remaining _.deburr() usages in the same file in favor of remove-accents that we've been using in other packages already.

This comes at the small cost in `wp-block-editor`:

Screenshot 2022-07-18 at 09 25 04

However, in the context of WordPress core it gets multiplied by every package/entry point that uses it:

#41865 (`wp-components`)

Screenshot 2022-07-18 at 09 26 23


#42429 (`wp-editor`)

Screenshot 2022-07-18 at 09 28 30

`lodash` size is substantial in contrast to a single `remove-accents`. However 3 bundled copies of `remove-accents` it's over 20% of `lodash`'s size. 

Screenshot 2022-07-18 at 09 32 47

**Comment by tyxla** [comment]
Thanks for the feedback @gziolo! 

As we progress with the Lodash migration, including a package multiple times is an inevitable side effect. However, it could mostly be addressed as we identify which those are, and then we could externalize some of them, and introduce them as vendor scripts in the WP core. I think this makes sense as part of #17025 and I can see you already brought those points up there. 

In any case, I think we should be able to go ahead with this one, and deal with the multiplication problem separately, resting assured that it's under our radar. I'm happy to devote some time to that, and including to some of the other packages that occur often (like `classname` for example). WDYT?

File: packages/block-editor/src/components/inserter/search-items.js
Comment (tyxla) [reply packages/block-editor/src/components/inserter/search-items.js:4]: Thanks for the feedback - FWIW `_.words()` is pretty costly, and while I haven't directly compared the costs with `noCase()`, I'm almost certain that `noCase()` will be cheaper. Can thoroughly compare if you have doubts, though.

Did you have any other reservations with the changes suggested in this PR, @talldan?

**Comment by gziolo** [comment]
@tyxla, you probably didn't notice that I edited my comment to make it clear that I don't think it's a blocker for this PR. 6kb is still less than 24kb from `lodash`, but it's something that we should constantly monitor so we maximize the benefits of full removal of `lodash` at some point.

**Comment by tyxla** [comment]
> @tyxla, you probably didn't notice that I edited my comment to make it clear that I don't think it's a blocker for this PR. 6kb is still less than 24kb from `lodash`, but it's something that we should constantly monitor so we maximize the benefits of full removal of `lodash` at some point.

👍 Fully agree on this one - it's part of the journey - as we iterate and progress with the migration, opportunities for optimization will arise, and we should be consciously grasping them. Again, thanks for bringing that up, awareness is the game changer here.

File: packages/block-editor/src/components/inserter/search-items.js
Comment (talldan) [reply packages/block-editor/src/components/inserter/search-items.js:4]: You're all good. This was just a passing comment while I was reviewing https://github.com/WordPress/gutenberg/pull/42459, which discussed this file.

**Comment by tyxla** [comment]
@gziolo @talldan and everyone else: considering that deduplication and externalization of packages can (and will) be done separately as part of  https://github.com/WordPress/gutenberg/issues/17025, and it shouldn't block this one, can I get another review of this PR? 

Thanks in advance 🙌

File: packages/block-editor/src/components/inserter/search-items.js
Comment (sgomes) [inline packages/block-editor/src/components/inserter/search-items.js:47]: Nit: I would prefer `.filter( s => !!s )`, `.filter( s => Boolean( s ) )`, or `.filter( s => s !== '' )`, as I find them much more readable, but I concede that `.filter( Boolean )` usage is pretty widespread, and has probably become an idiom at this point.

**Review by sgomes — APPROVED** [review]
LGTM, thanks @tyxla! 👍

File: packages/block-editor/src/components/inserter/search-items.js
Comment (tyxla) [reply packages/block-editor/src/components/inserter/search-items.js:47]: Yeah, we're doing that in quite a bunch of places in Gutenberg already, and it's already become idiomatic in this codebase IMHO.

**Comment by tyxla** [comment]
This one was partially reverted in #44652, but take 2 is here: #45281.

**Pull request merged**
